pkg/machine/qemu/options_darwin_arm64.go

```diff
@@ -46,13 +46,23 @@ func getOvmfDir(imagePath, vmName string) string {
  */
 func getEdk2CodeFdPathFromQemuBinaryPath() string {
 	cfg, err := config.Default()
-	if err == nil {
-		execPath, err := cfg.FindHelperBinary(QemuCommand, true)
-		if err == nil {
-			return filepath.Clean(filepath.Join(filepath.Dir(execPath), "..", "share", "qemu"))
-		}
+	if err != nil {
+		return ""
+	}
+	execPath, err := cfg.FindHelperBinary(QemuCommand, true)
+	if err != nil {
+		return ""
+	}
+
+	sharePath := func(path string) string {
+		return filepath.Clean(filepath.Join(filepath.Dir(path), "..", "share", "qemu"))
+	}
+
+	symlinkedPath, err := filepath.EvalSymlinks(execPath)
```

Comment:
This should probably be done in the containers/common cfg.FineHelperBinary() so that all callers get the benefit.

Reply:
Thanks for your input!

I'd be happy to make a PR there, though I hesitate to apply it so broadly, as resolving the symlink could change behavior in some cases; I thought it was safe and reasonable in this case as we're looking specifically for the `qemu-system-aarch64` binary, and only on aarch64-darwin, and this seems to solve a semi-common issue for nix + podman users on this specific platform.

For example, resolving the symlink could change the returned filename in unexpected ways (which I think is unlikely to be a problem with `qemu-system-aarch64`, but could in other contexts).

```go
package main

import (
	"os"
	"path/filepath"
)

func main() {
	cwd, _ := os.Getwd()
	path, name := cwd, "fake-link"
	fullpath := filepath.Join(path, name)
	if fi, err := os.Stat(fullpath); err == nil && fi.Mode().IsRegular() {
		println("no error", fullpath)

		if p, err := filepath.EvalSymlinks(fullpath); err == nil {
			println("resolved", p)
			return
		}
		return
	}
	println("whups")
}
```

```console
$ ln -s $(which qemu-system-aarch64) ./fake-link
$ go run .
no error /var/folders/kb/tw_lp_xd2_bbv0hqk4m0bvt80000gn/T/tmp.sttNLyt8Xi/fake-link
resolved /nix/store/b5msl7ib515wdfvgvcxl370x1p1p7s4r-qemu-7.2.0/bin/qemu-system-aarch64
```

Reply:
@rhatdan PTAL